### ISSUE TITLE
YAML ruleset format not recognised if ruleset URL does not end with .yml

### ISSUE THREAD

**Issue by ngayemdm** [stoplightio/spectral#1918]
**Describe the bug**
When fetching a custom YAML ruleset by an URL not ending with .yml, spectral misinterprets the format and returns an error message like "Unexpected token (Note that you need plugins to import files that are not JavaScript)"

**To Reproduce**

1. Place the openapi file in this archive [contract.openapi.yml.zip](https://github.com/stoplightio/spectral/files/7432259/contract.openapi.yml.zip) in the current folder
2. Run this CLI command 
```
docker run --rm -v $(pwd):/tmp:ro stoplight/spectral:6.1 --ruleset "https://gist.githubusercontent.com/ngayemdm/2bae1bc0083a4413b427fbeeee4139be/raw/ed9b90807ffec2750fafa
9e7c6f48eeb2b01e7d4/customer-ruleset.yml?fake_private_token=qwerty" lint /tmp/contract.openapi.yml  -v
```
3. See error

**Expected behavior**
The ruleset should be detected as a YAML file and the OpenAPI file should be analysed.

**Environment (remove any that are not applicable):**
 - Library version: 6.1.0 (using Docker image)
 - OS: Ubuntu 20.04.2 LTS

**Additional context**
No problems detected when using the docker image stoplight/spectral:5.9 with command 
```
docker run --rm -v $(pwd):/tmp:ro stoplight/spectral:5.9 --ruleset "https://gist.githubusercontent.com/ngayemdm/2bae1bc0083a4413b427fbeeee4139be/raw/ed9b90807ffec2750fafa
9e7c6f48eeb2b01e7d4/customer-ruleset.yml?fake_private_token=qwerty" lint /tmp/contract.openapi.yml  -v
```

**Comment by P0lip** [comment]
Hey!
To be honest, that's expected behavior.
If a ruleset doesn't end up with `.yaml`, `.yml` or `.json` it's considered a JS ruleset, and hence such a ruleset should be provided.
We could probably get `Content-Type` to see what the output is in case we cannot determine it based on the extension.
Is there any reason you cannot use the filename containing an extension?

**Comment by ngayemdm** [comment]
Hello, 
I use Gitlab to store my rulesets in a dedicated project. 
The raw file is accessible via the Gitlab API and the URL has this format: `https://<my_company_hostname>/api/v4/projects/<gitlabl_project_id>/repository/files/rules%2Foas.spectral.yml/raw?private_token=<read_only_token>&ref=<main_branch_of_my_project>`

The extension is present but not at the end of the URL.
The `/raw` part is required to get the content of the file instead of the metadata.
And i have to use query parameters to handle the authentication and the git ref (branch or tag).

If i add a dummy param like `&dummy=.yml` at the end of the URL, spectral will correctly interpret the file as a YAML.

**Comment by P0lip** [comment]
Got it. If it's served with a correct `Content-Type` we might be able to use that to determine the format of the ruleset.
However, if it's served as plain text, you'll probably need to continue using that dummy param.

I'll keep you posted - hoping to add that `Content-Type` based detection at some point.

**Comment by ngayemdm** [comment]
For security reasons, gitlab uses the `Content-Type` "`text/plain`" when returning the contents of my file.
Thanks for your comments, I will continue to use my setting for now.
